dist/esm5/common/ICacheBusterConfig.d.ts

```diff
@@ -6,3 +6,11 @@ export interface ICacheBusterConfig {
      */
     cacheBusterNotifier?: Subject<void>;
 }
+/**
+ * flag that indicates whether cache should be cleared before decorated method has been called.
+ * False by default, i.e. cache is cleared after observable from decorated method emits
+ */
+export declare type ICacheBusterConfigInstant = ICacheBusterConfig & {
+    isInstant: true;
```

Comment:
That was my first take, but we lost typing then. This way it was possible to decorate a method that did not return Observable/Promise with @CacheBuster (with non instant flag). With different type we can enforce that using function overload. So now only if you set isInstant: true, the typing will be loose, i.e it can decorate any method, but if you don`t specify isInstant, then only Observable/Promise.

Reply:
Or is it really? I think we could use generics there... Ill give it a try.